src/main/java/org/takes/misc/Utf8String.java

```diff
@@ -30,6 +30,11 @@
  * @author Maksimenko Vladimir (xupypr@xupypr.com)
  * @version $Id$
  * @since 0.33
+ * @todo #804:30min Make this implements Cactoos Bytes and Text
```

Comment:
@victornoel it's not correct, if we name this class as utf-8 string, we can't treat it as `Bytes`. We should either rename this class or remove todo to implement `Bytes`

Reply:
@g4s8 the class has a method `bytes`, it is part of its behaviour, but it represents a `String`.
I can remove `Bytes` but I think it would a pity, and I have no idea for another name that makes sense. What do you think?

Reply:
@victornoel maybe it's better to call it `Utf8Bytes`, because it's main behaviour:
```java
     public byte[] bytes() {
         return this.value.getBytes(Charset.forName(Utf8String.ENCODING));
}
```

WDYT?

Reply:
@g4s8 the thing I can't wrap my head around is that if I have a `Bytes` and I use `TextOf` to convert it to `Text`, I need to provide a `Charset` to `TextOf` to be sure of the encoding. And if I have a `Text` and I use `BytesOf` to convert it to `Bytes`, I need to provide a `Charset` to `BytesOf` to be sure of the encoding.

So if I want to be sure that I am manipulating an UTF8 string, I HAVE to use an object that is both implementing `Bytes` and `Text`, if not I can't use safely `TextOf` or `BytesOf` for the conversion.

See what I mean? If we name it `Utf8Bytes`, it's good to me, but in the end, it's also an `Utf8String`... And from a conceptual point of view, it's always a String before being an array of bytes...

Reply:
@victornoel right, it's a discussion for another ticket